server/historian/packages/historian-base/src/routes/utils.ts

```diff
@@ -118,18 +123,36 @@ export async function createGitService(createArgs: createGitServiceArgs): Promis
 	const customData: ITenantCustomDataExternal = details.customData;
 	const writeToExternalStorage = !!customData?.externalStorageData;
 	const storageUrl = config.get("storageUrl") as string | undefined;
+	const ignoreEphemeralFlag: boolean = config.get("ignoreEphemeralFlag");
 	const maxCacheableSummarySize: number =
 		config.get("restGitService:maxCacheableSummarySize") ?? 1_000_000_000; // default: 1gb
 
-	let isEphemeral = cache ? isEphemeralContainer : false;
+	let isEphemeral: boolean = isEphemeralContainer;
 	if (!ignoreEphemeralFlag) {
-		if (isEphemeralContainer !== undefined) {
-			await cache?.set(`isEphemeral:${documentId}`, isEphemeralContainer);
+		const isEphemeralKey: string = `isEphemeralContainer:${documentId}`;
+		if (isEphemeral !== undefined && isEphemeral !== null) {
+			// If an isEphemeral flag was passed in, cache it in Redis
+			Lumberjack.info(`Setting ${isEphemeralKey} to ${isEphemeral}`);
+			await cache?.set(isEphemeralKey, isEphemeral);
 		} else {
-			isEphemeral = await cache?.get(`isEphemeral:${documentId}`);
-			// Todo: If isEphemeral is still undefined fetch the value from database
+			isEphemeral = await cache?.get(isEphemeralKey);
+			if (isEphemeral === null || isEphemeral === undefined) {
```

Comment:
Is this equivalent to if (!isEphemeral)

Reply:
The difference here is similar to the other one. I don't want it to fallback to the document static data if there is an actual value for isEphemeral stored in the cache (either true or false). But doing (!isEphemeral) would evaluate to true if isEphemeral=false, which would make it go to the document read even though we already know the document is not ephemeral. This syntax is used so that it will only use the backup when there was nothing in the cache (null/undefined)

Reply:
Ended up simplifying this a bit by making it be isEphemeral == undefined. Was a bit reluctant to do at first, because I have to disable an eslint check to allow it, but I think it will be cleaner and it avoids one of the warnings about no explicit null values :)